packages/common-scripts/src/omnilock.ts

```diff
@@ -345,10 +373,17 @@ export async function setupInputCell(
       );
     }
     let witness: string = txSkeleton.get("witnesses").get(firstIndex)!;
+
+    const placeholderLength =
+      bytes.bytify(inputCell.cellOutput.lock.args)[0] ===
+      IdentityFlagsType.IdentityFlagsSolana
+        ? ED25519_SIGNATURE_PLACEHOLDER_LENGTH
+        : SECP256K1_SIGNATURE_PLACEHOLDER_LENGTH;
```

Comment:
It's easy to be a bug if it treats `SECP256K1_SIGNATURE_PlACEHOLDER_LENGTH` as a fallback, should be exactly matched or throw an error when none is matched.